pkg/apis/camel/v1/integrationprofile_types.go

```diff
@@ -0,0 +1,138 @@
+}
+
+// +kubebuilder:object:root=true
+
+// IntegrationProfileList contains a list of IntegrationProfile.
+type IntegrationProfileList struct {
+	metav1.TypeMeta `json:",inline"`
+	metav1.ListMeta `json:"metadata,omitempty"`
+	Items           []IntegrationProfile `json:"items"`
+}
+
+// IntegrationProfileBuildSpec contains profile related build information.
+// This configuration can be used to tune the behavior of the Integration/IntegrationKit image builds.
+type IntegrationProfileBuildSpec struct {
+	// the Camel K Runtime dependency version
+	RuntimeVersion string `json:"runtimeVersion,omitempty"`
+	// the runtime used. Likely Camel Quarkus (we used to have main runtime which has been discontinued since version 1.5)
+	RuntimeProvider RuntimeProvider `json:"runtimeProvider,omitempty"`
+	// a base image that can be used as base layer for all images.
+	// It can be useful if you want to provide some custom base image with further utility software
+	BaseImage string `json:"baseImage,omitempty"`
+	// the image registry used to push/pull Integration images
+	Registry RegistrySpec `json:"registry,omitempty"`
+	// how much time to wait before time out the pipeline process
+	Timeout *metav1.Duration `json:"timeout,omitempty"`
+	// Maven configuration used to build the Camel/Camel-Quarkus applications
+	Maven MavenSpec `json:"maven,omitempty"`
+}
+
+// IntegrationProfileKameletSpec define the behavior for all the Kamelets controller by the IntegrationProfile.
+type IntegrationProfileKameletSpec struct {
+	// remote repository used to retrieve Kamelet catalog
+	Repositories []KameletRepositorySpec `json:"repositories,omitempty"`
+}
+
+// IntegrationProfilePhase is the phase of an IntegrationProfile.
+type IntegrationProfilePhase string
+
+// IntegrationProfileConditionType defines the type of condition.
+type IntegrationProfileConditionType string
+
+const (
+	// IntegrationProfileKind is the Kind name of the IntegrationProfile CR.
+	IntegrationProfileKind string = "IntegrationProfile"
+
+	// IntegrationProfilePhaseNone when the IntegrationProfile does not exist.
+	IntegrationProfilePhaseNone IntegrationProfilePhase = ""
+	// IntegrationProfilePhaseReady when the IntegrationProfile is ready.
+	IntegrationProfilePhaseReady IntegrationProfilePhase = "Ready"
+	// IntegrationProfilePhaseError when the IntegrationProfile had some error (see Conditions).
+	IntegrationProfilePhaseError IntegrationProfilePhase = "Error"
+
+	// IntegrationProfileConditionTypeCreated is the condition if the IntegrationProfile has been created.
+	IntegrationProfileConditionTypeCreated IntegrationProfileConditionType = "Created"
+
+	// IntegrationProfileConditionCreatedReason represents the reason that the IntegrationProfile is created.
+	IntegrationProfileConditionCreatedReason = "IntegrationProfileCreated"
+)
+
+// IntegrationProfileCondition describes the state of a resource at a certain point.
+type IntegrationProfileCondition struct {
```

Comment:
nitpick: I know we have done this for may of the types but wonder if we can just reuse `metav1.Condition`, which would allow to manipulate conditions using some [k8s utilities](https://github.com/kubernetes/apimachinery/blob/master/pkg/api/meta/conditions.go). The downside is that we won't have the specific condition type, but given 99% of the time is just an alias to strings and there are no real enums in go, I don't see anymore a very string benefit.

Reply:
Thanks for the idea. I have created an issue for this #5191